## What factors impact heat related hospitalizations in NYC?

<B>Organized by community district (CD), I'm looking at facets of climate and socioeconomic data to see what associations exist

#Keys: Percent = percentage of land with vegetative structures; Degrees = Daytime summer surface temperature; Score = Heat Vulnerability Index; HospRate = Age adjusted rate of heat related hospitalizations per 100,000
    

In [74]:
#Loading all the things!

In [75]:
%load_ext rpy2.ipython
%load_ext autoreload
%autoreload 2

%matplotlib inline  
from matplotlib import rcParams
rcParams['figure.figsize'] = (16, 100)

import warnings
from rpy2.rinterface import RRuntimeWarning
warnings.filterwarnings("ignore") # Ignore all warnings
# warnings.filterwarnings("ignore", category=RRuntimeWarning) # Show some warnings

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from IPython.display import display, HTML

The rpy2.ipython extension is already loaded. To reload it, use:
  %reload_ext rpy2.ipython
The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [76]:
%%javascript
// Disable auto-scrolling
IPython.OutputArea.prototype._should_scroll = function(lines) {
    return false;
}

<IPython.core.display.Javascript object>

In [77]:
import pandas as pd
df = pd.read_csv("sample.csv") #this has some merging from my scratchpad
dfHosp = pd.read_csv("CD_Hospitalizations.csv")

df2 = pd.merge(df, dfHosp, on='Geography', how='left')
df2

GeoType                                       Geography  Percent  Degrees  \
0       CD                        Financial District (CD1)     21.9       94   
1       CD                Greenwich Village and Soho (CD2)     15.3       98   
2       CD             Lower East Side and Chinatown (CD3)     25.9       98   
3       CD                       Clinton and Chelsea (CD4)     12.8       98   
4       CD                                   Midtown (CD5)      4.7       98   
5       CD            Stuyvesant Town and Turtle Bay (CD6)     20.8       97   
6       CD                           Upper West Side (CD7)     28.9       96   
7       CD                           Upper East Side (CD8)     21.7       97   
8       CD  Morningside Heights and Hamilton Heights (CD9)     32.4       95   
9       CD                           Central Harlem (CD10)     24.3       98   
10      CD                              East Harlem (CD11)     32.6       97   
11      CD            Washington Heights and Inwood (CD12)     37.7       94   
12      CD                    Mott Haven and Melrose (CD1)     19.8      101   
13      CD                  Hunts Point and Longwood (CD2)     13.5      101   
14      CD                    Morrisania and Crotona (CD3)     30.1       99   
15      CD                  Highbridge and Concourse (CD4)     22.2       98   
16      CD            Fordham and University Heights (CD5)     20.3       98   
17      CD                  Belmont and East Tremont (CD6)     21.8       99   
18      CD           Kingsbridge Heights and Bedford (CD7)     24.7       97   
19      CD                   Riverdale and Fieldston (CD8)     48.9       93   
20      CD                 Parkchester and Soundview (CD9)     32.0       98   
21      CD               Throgs Neck and Co-op City (CD10)     39.0       97   
22      CD                Morris Park and Bronxdale (CD11)     29.7      100   
23      CD            Williamsbridge and Baychester (CD12)     38.5       98   
24      CD               Greenpoint and Williamsburg (CD1)     13.9      100   
25      CD          Fort Greene and Brooklyn Heights (CD2)     24.6       98   
26      CD                        Bedford Stuyvesant (CD3)     25.1       99   
27      CD                                  Bushwick (CD4)     25.4      100   
28      CD           East New York and Starrett City (CD5)     28.3      100   
29      CD            Park Slope and Carroll Gardens (CD6)     20.8       97   
30      CD                               Sunset Park (CD7)     29.4       96   
31      CD        Crown Heights and Prospect Heights (CD8)     24.6       98   
32      CD  South Crown Heights and Lefferts Gardens (CD9)     21.5       98   
33      CD              Bay Ridge and Dyker Heights (CD10)     32.3       94   
34      CD                              Bensonhurst (CD11)     15.6       96   
35      CD                             Borough Park (CD12)     23.9       99   
36      CD                             Coney Island (CD13)     22.8       96   
37      CD                     Flatbush and Midwood (CD14)     29.3       99   
38      CD                           Sheepshead Bay (CD15)     23.4       98   
39      CD                              Brownsville (CD16)     23.1      101   
40      CD                            East Flatbush (CD17)     22.0       99   
41      CD                   Flatlands and Canarsie (CD18)     34.9       98   
42      CD              Long Island City and Astoria (CD1)     23.2       99   
43      CD                    Woodside and Sunnyside (CD2)     23.4      100   
44      CD                           Jackson Heights (CD3)     24.3      100   
45      CD                       Elmhurst and Corona (CD4)     18.3      101   
46      CD                     Ridgewood and Maspeth (CD5)     38.2       98   
47      CD                Rego Park and Forest Hills (CD6)     34.5       99   
48      CD                   Flushing and Whitestone (CD7)     38.1       99   
49 

In [78]:
dfPoverty = pd.read_csv("CD_Poverty.csv")
dfCombined = pd.merge(df2, dfPoverty, on='Geography', how='left')

In [79]:
%%R
# load the dataset
data(dfCombined)

# run a simple linear regression of mpg (dependent variable) on wt (independent variable)
fit <- lm("HospRate ~ Percent_Poverty + Percent + Score + Degrees" , data = dfCombined)

# summarize the regression results
summary(fit)





Call:
lm(formula = "HospRate ~ Percent_Poverty + Percent + Score + Degrees", 
    data = dfCombined)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.89753 -0.29663 -0.03548  0.18593  1.33331 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)   
(Intercept)      3.129018   4.573313   0.684  0.49678   
Percent_Poverty  0.029384   0.011720   2.507  0.01521 * 
Percent          0.020656   0.007357   2.808  0.00693 **
Score            0.271492   0.098929   2.744  0.00821 **
Degrees         -0.036416   0.047477  -0.767  0.44641   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5434 on 54 degrees of freedom
Multiple R-squared:  0.5205,	Adjusted R-squared:  0.485 
F-statistic: 14.65 on 4 and 54 DF,  p-value: 3.628e-08



In [80]:
%%R
# load the dataset
data(dfCombined)


# run a simple linear regression of mpg (dependent variable) on wt (independent variable)
fit <- lm("HospRate ~ Degrees" , data = dfCombined)

# summarize the regression results
summary(fit)



Call:
lm(formula = "HospRate ~ Degrees", data = dfCombined)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.2114 -0.5619 -0.0377  0.4512  1.8643 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)
(Intercept) -5.81421    4.74884  -1.224    0.226
Degrees      0.07475    0.04842   1.544    0.128

Residual standard error: 0.7484 on 57 degrees of freedom
Multiple R-squared:  0.04013,	Adjusted R-squared:  0.02329 
F-statistic: 2.383 on 1 and 57 DF,  p-value: 0.1282



<B>Shows me that its very clearly not just surface temperature that impacts Heat related hospitalizations

In [83]:
%%R
# load the dataset
data(dfCombined)


# run a simple linear regression of mpg (dependent variable) on wt (independent variable)
fit <- lm("HospRate ~ Percent + Score + Degrees" , data = dfCombined)

# summarize the regression results
summary(fit)


Call:
lm(formula = "HospRate ~ Percent + Score + Degrees", data = dfCombined)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.92396 -0.35177 -0.04469  0.24321  1.53401 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  7.557670   4.416500   1.711  0.09267 .  
Percent      0.021747   0.007689   2.828  0.00651 ** 
Score        0.455931   0.069252   6.584 1.81e-08 ***
Degrees     -0.081918   0.045931  -1.783  0.08003 .  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.569 on 55 degrees of freedom
Multiple R-squared:  0.4646,	Adjusted R-squared:  0.4354 
F-statistic: 15.91 on 3 and 55 DF,  p-value: 1.438e-07



In [86]:
%%R
# load the dataset
data(dfCombined)


# run a simple linear regression of mpg (dependent variable) on wt (independent variable)
fit <- lm("HospRate ~ I(Percent**2) + I(Score**2) + I(Degrees**2) + I(Percent_Poverty**2)" , data = dfCombined)

# summarize the regression results
summary(fit)


Call:
lm(formula = "HospRate ~ I(Percent**2) + I(Score**2) + I(Degrees**2) + I(Percent_Poverty**2)", 
    data = dfCombined)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.76416 -0.28452 -0.07733  0.19512  1.51916 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)   
(Intercept)           2.7519541  2.2619537   1.217  0.22904   
I(Percent^2)          0.0002499  0.0001093   2.285  0.02624 * 
I(Score^2)            0.0508905  0.0156757   3.246  0.00201 **
I(Degrees^2)         -0.0002329  0.0002394  -0.973  0.33506   
I(Percent_Poverty^2)  0.0004697  0.0002617   1.795  0.07827 . 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5609 on 54 degrees of freedom
Multiple R-squared:  0.4892,	Adjusted R-squared:  0.4513 
F-statistic: 12.93 on 4 and 54 DF,  p-value: 1.89e-07



In [87]:
%%R
# load the dataset
data(dfCombined)


# run a simple linear regression of mpg (dependent variable) on wt (independent variable)
fit <- lm("Score ~ I(Percent**2) + I(Degrees**2) + I(Percent_Poverty**2)" , data = dfCombined)

# summarize the regression results
summary(fit)


Call:
lm(formula = "Score ~ I(Percent**2) + I(Degrees**2) + I(Percent_Poverty**2)", 
    data = dfCombined)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.57346 -0.56134 -0.08567  0.52658  2.02489 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)    
(Intercept)          -1.142e+01  2.743e+00  -4.165 0.000111 ***
I(Percent^2)         -3.749e-04  1.513e-04  -2.479 0.016282 *  
I(Degrees^2)          1.454e-03  2.821e-04   5.154 3.57e-06 ***
I(Percent_Poverty^2)  1.782e-03  2.536e-04   7.026 3.41e-09 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.7938 on 55 degrees of freedom
Multiple R-squared:  0.7112,	Adjusted R-squared:  0.6955 
F-statistic: 45.15 on 3 and 55 DF,  p-value: 7.455e-15



<B>wait woah that's much higher than before -- seeing what factors make for high heat vulnerability index

In [95]:
%%R
# load the dataset
data(dfCombined)


# run a simple linear regression of mpg (dependent variable) on wt (independent variable)
fit <- lm("HospRate ~ I(log(Percent)) + I(log(Score)) + I(log(Degrees)) + I(log(Percent_Poverty))" , data = dfCombined)

# summarize the regression results
summary(fit)


Call:
lm(formula = "HospRate ~ I(log(Percent)) + I(log(Score)) + I(log(Degrees)) + I(log(Percent_Poverty))", 
    data = dfCombined)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.09528 -0.33697 -0.07567  0.32184  1.41104 

Coefficients:
                        Estimate Std. Error t value Pr(>|t|)   
(Intercept)               3.7813    22.1594   0.171  0.86514   
I(log(Percent))           0.4665     0.1974   2.363  0.02174 * 
I(log(Score))             0.4503     0.2471   1.822  0.07395 . 
I(log(Degrees))          -1.3288     4.8205  -0.276  0.78386   
I(log(Percent_Poverty))   0.6667     0.2110   3.159  0.00259 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5628 on 54 degrees of freedom
Multiple R-squared:  0.4857,	Adjusted R-squared:  0.4477 
F-statistic: 12.75 on 4 and 54 DF,  p-value: 2.246e-07



## Interesting to me that poverty is what changed it the most ??? Time to make some calls and maybe test some more things!

<B> .45 for my adjusted R squared isn't bad but it's not super high either, though we have a low p value so it's not likely that this is due to chance alone 